### PR TITLE
Add local settings docs

### PR DESCRIPTION
**Pull request by ealeksandrov**
## Description

This PR adds documentation for different types of local data stored in `AppSettingsStore`.
Internal discussion ref: p91TBi-6pI-p2.

---
Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: docs/app-local-settings.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# App Local Settings
+
+Small amount of data is currently stored in plist files.
```
Comment (rachelmcr) [inline docs/app-local-settings.md]: nit: grammar

```suggestion
A small amount of data is currently stored in plist files.
```

File: docs/app-local-settings.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Initial goals were:
+
+- prevent complexity overhead of Core Data
+- prevent data loss when we auto-create the database (if a migration fails)
+- make data typed
+- inject storage layer as dependency.
+
+Retreiving and storing the data is done via [`AppSettingsAction`](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/Yosemite/Actions/AppSettingsAction.swift) and most of logic happens in [`AppSettingsStore`](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/Yosemite/Stores/AppSettingsStore.swift).
```
Comment (rachelmcr) [inline docs/app-local-settings.md]: nit: spelling/grammar

```suggestion
Retrieving and storing the data is done via [`AppSettingsAction`](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/Yosemite/Actions/AppSettingsAction.swift) and most of the logic happens in [`AppSettingsStore`](https://github.com/woocommerce/woocommerce-ios/blob/develop/Yosemite/Yosemite/Stores/AppSettingsStore.swift).
```

File: docs/app-local-settings.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+There are few data models + plist files separated for specific features and use cases.
```
Comment (rachelmcr) [inline docs/app-local-settings.md]: nit: grammar

```suggestion
There are a few data models + plist files separated for specific features and use cases.
```

**Review by rachelmcr — APPROVED** [review]
This is really helpful; thanks for documenting it! I left some small suggested changes (all nits) but the documentation is clear and concise. 👍

**Comment by ealeksandrov** [comment]
Thanks for review and corrections 👍

**Pull request merged**